tests/rptest/tests/topic_creation_test.py

```diff
@@ -466,25 +466,62 @@ def test_retention_config_on_upgrade_from_v22_2_to_v22_3(self):
         # later for checking that deletion behavior is correct on legacy topics.
         self._populate_tiered_storage_topic("test-si-topic-with-retention")
 
+        # This topic is like "test-si-topic-with-retention", but instead
+        # of settings its properties at creation time, set them via
+        # alter messages: these follow a different code path in
+        # kafka and controller, and are serialized using different structures
+        # than the structures used in topic creation.
+        self.rpk.create_topic("test-si-topic-with-retention-altered")
+        self.rpk.alter_topic_config("test-si-topic-with-retention-altered",
+                                    "retention.bytes", 10000)
+        self.rpk.alter_topic_config("test-si-topic-with-retention-altered",
+                                    "redpanda.remote.write", "true")
+
+        # Check our alter operations applied properly
+        # (see https://github.com/redpanda-data/redpanda/issues/6772)
+        after_alter = self.rpk.describe_topic_configs(
+            "test-si-topic-with-retention-altered")
+        assert after_alter['redpanda.remote.write'][0] == 'true'
+        assert after_alter['retention.bytes'][0] == '10000'
+
+        # TODO: this test currently exercises 22.2 (serde) encoding to
+        # 22.3 (newer serde) encoding.  It should be extended to also
+        # cover the case of creating topics in version 22.1, upgrading
+        # to 22.2, and then to 22.3, to check the ADL encoding of messages
+        # in the controller log.
+
         self.installer.install(
             self.redpanda.nodes,
             RedpandaInstaller.HEAD,
         )
 
         self.redpanda.restart_nodes(self.redpanda.nodes)
 
+        # TODO: wait for features to all activate
+        admin = Admin(self.redpanda)
+        sleep(10)
```

Comment:
nit: Can we rewrite this as a `wait_until`? We've had a few `sleeps` slip through into our tests recently and I don't think that's a good direction to head in.

Reply:
:facepalm: ignoring my own TODO!  Fixed.